### ISSUE TITLE
Window Splitting Feature?

### ISSUE THREAD

**Issue by chrooted-mother** [hexchat/hexchat#34]
I use Hexchat on my Windows OS and it's lovely, my favorite irc client. There's one thing I really miss though when I'm using it instead of weechat though. Window Splitting. I'm not sure how hard it would be to add this feature but I'd really love if you could add it.

Possibly set it up so you can have multiple pages of split windows so you can have like Page 1 with your channels in Server A, Page 2 with channels in Server B? Just having something like weechat or irssi would still make me very happy though.

Thanks!

**Comment by gdude2002** [comment]
Just putting my vote in for this. I'm getting tired of mIRC and HexChat seems like a fantastic client, but I simply can't use IRC efficiently without my subwindows.

> ...does that mean we would get title bars and overlapping channels all over the place?

mIRC has tiling options and stuff, you can easily [make them tile automatically](http://i.imgur.com/zUxTEbx.png).

**Comment by Poorchop** [comment]
With all due respect, that UI looks awful and there is so much wasted space, which is why pane splitting would be much better. Only a single input bar is needed, as well as a single user list that changes based on the active context. As mentioned above, you can get that exact same layout in your picture if you just detach channels from the initial HexChat window and tile them yourself.

**Comment by gdude2002** [comment]
> ...you can get that exact same layout in your picture if you just detach channels from the initial HexChat window and tile them yourself

No, these are MDI windows - they tile automatically in mIRC, and are contained within one main window that I can simply leave maximized on its own screen. Split windows are extremely painful to manage.

> ...that UI looks awful and there is so much wasted space

Agreed. But no less wasted space than actual separate windows.

**Comment by Corrodias** [comment]
Indeed. It would be preferable to have the option to make the child windows titleless or even borderless, but failing that, it's still better than separate windows because you can have the common toolbar elements and whatnot in the main window.

**Comment by circuit-breaker** [comment]
I know MDI is the 'red-headed stepchild' of the widget/UI toolkit world, but this is the _sole_ use case I can think where I actually WANT it.  It serves a very functional purpose in this specific case, that otherwise must be handled outside the client. Unless that's done right, it'll be quite awkward to manage for those of us with lots of channels to tile, remove borders from.. and that's for a static view.

MDI still haunts us via Office (that's the one I know of) so let's at least use it to our advantage, maybe?

Does the GTK toolkit support such a thing?

**Comment by gdude2002** [comment]
As far as I know, GTK does not support MDI windows.

**Comment by gdude2002** [comment]
Ah, looks like you edited your comment. Well..

---

Tiling is not a full solution to this issue. It's definitely something that should be included, though - see this gif for what mIRC does.

![Example of MDI in mIRC](https://dl.dropboxusercontent.com/u/1547120/ShareX/2016/May/2016-05-18_10-56-45.gif)
[Full image link](https://dl.dropboxusercontent.com/u/1547120/ShareX/2016/May/2016-05-18_10-56-45.gif)

As you can see, this is more than just tiling - you can maximize windows or hide them at your leisure. Additionally, if you don't want to tile, you don't have to - you can arrange the windows any way you please. If it's possible to do this, then I feel like we should default to preferring it, as it is nowhere near as limited as simple tiling with borders.

Additionally, Hexchat doesn't appear to support hiding the treebar (list of channels to the left), so that support would need to exist for the tiling setup.

**Comment by circuit-breaker** [comment]
Thanks for the reply, and sorry for the ninja edit.. :) I just had realized what I liked and felt worked w/the terminal examples

**Comment by Corrodias** [comment]
This is *the* reason I still use mIRC instead of anything else. I'm just glad it's back under development and is adding IRCv3 features, so it plays nicely with ZNC's server timestamps.

**Comment by gdude2002** [comment]
Unfortunately I doubt it'll happen. To my knowledge, GTK still doesn't
support what would be required - and with the way that project is going,
it probably never will.

---
This email has been checked for viruses by Avast antivirus software.
https://www.avast.com/antivirus

**Comment by Corrodias** [comment]
It turns out KVIrc removed MDI and can no longer do this. mIRC is now the only client with this capability.

**Comment by circuit-breaker** [comment]
GTK definitely doesn't look like it'll *ever* add anything like MDI, per what was said before.  This is a pipe dream, however ill-advised MDI is - we were willing to make concessions for this single use-case.. but MDI over-all is not worth GTK dev time, no matter how much we just have to have our single window with tiled irc channels.  Close?

**Comment by Corrodias** [comment]
Understood, and I can't blame the team for prioritizing as it has. I don't necessarily want MDI - in fact I don't particularly like how it looks - however, it's clearly possible to create different... panels/frames, as KVIrc has channel list, the channel content, the name list, and so on. Is it conceivable that, some time in the future, it could display more than one channel in a grid layout using those mechanisms instead?

At present, my unrealistic dream is that someone makes an HTML/JS-based client that renders its own, movable windows inside, but an implementation of a new IRC client from the ground up, with such an esoteric feature, isn't likely to happen.

**Comment by circuit-breaker** [comment]
I think you're entirely right that that is how it will do it, and I hope it also displays image media inline... that'd be really neat.  I wouldn't mind that one bit except for the memory utilization.  Life is made of tiny concessions though..

**Comment by learn-more** [comment]
It does not have to be MDI?
Split views would already be a solution to this imo.

**Comment by gdude2002** [comment]
Split views would work fine for me as long as they're within the same overall window. There'd be a few things to consider though, eg the mechanics of switching channels and whether each view has its own input

**Comment by JohnRobson** [comment]
1+ for this feature.

**Comment by Hebgbs** [comment]
Pidgin does this! How the hell doesn't GTK support this when Pidgin and Gimp does tab splitting, joiing and arranging so well?

**Comment by TingPing** [comment]
@Hebgbs Gtk provides nothing built in like this. It is a general purpose and extensible toolkit of course an application can implement features on top of it.
